lib/yardstick/v2_client/remote_model.rb

```diff
@@ -40,7 +40,7 @@ def instance_uri
 
       module ClassMethods
         def find(token, id)
-          from_api(get("#{resource_uri}/#{id}", query: { token: token }))
+          from_api(get("#{resource_uri}/#{id}", query: { token: token }), token: token)
```

Comment:
The second token hash is the "extras" param for `from_api`. It's so the token get's assigned to the instance of the class returned by find, and then can be used in other methods on the object.